tests/unit/unit-walker.cpp

```diff
@@ -37,10 +39,67 @@ class UnitWalkerTests : public ::testing::Test,
 
     bvsort = s->make_sort(BV, 4);
     funsort = s->make_sort(FUNCTION, SortVec{ bvsort, bvsort });
+    fun2sort = s->make_sort(FUNCTION, SortVec{ bvsort, bvsort, bvsort });
     arrsort = s->make_sort(ARRAY, bvsort, bvsort);
+    boolsort = s->make_sort(BOOL);
   }
   SmtSolver s;
-  Sort bvsort, funsort, arrsort;
+  Sort bvsort, funsort, fun2sort, arrsort, boolsort;
+};
+
+/* Custom Tree Walker that builds up a map assigning fresh boolean variables to
+ * each occurrence of each term in the formula it visits. One example of
+ * overriding visit_term to customize the behavior of the walker.
+ * Class gets used in last 9 tests: SimpleTree, PathDecomp, PathTests1,
+ * SimplePathTests1, SimplePathTests2, PathTests3, PathTestsUF1, PathTestsUF2,
+ * FreshVars
+ */
+class IndicatorTreeWalker : public TreeWalker
+{
+  // counter with which to name fresh variables: b0, ..., bn
+  int b_iter{};
+
+  using TreeWalker::TreeWalker;
+
+  /* builds up map assigning fresh boolean variables to each occurrence of each
+   * term in the formula, where the occurrences are represented as a  pair
+   * containing the formula the term appears in and the path indicating the
+   * term's location in the formula it appears in. For example, the formula
+   * x+2=y traversed with this custom version of visit_term will build the
+   * corresponding map containing: b0 -> <x+2=y, []>, corresponding to the term
+   * x+2=y b1 -> <x+2=y, [1]>, corresponding to the term y b2 -> <x+2=y, [0]>,
+   * corresponding to the term x+2 b3 -> <x+2=y, [0,1]>, corresponding to the
+   * term 2 b4 -> <x+2=y, [0,0]>, corresponding to the term x As in this
+   * example, each occurrence of each term in the formula should correspond to a
+   * fresh variable mapping to its occurrence in the formula
```

Comment:
i guess clang-format deleted some new-lines here. Please re-format this part yourself, and in the next commits we will use git-clang-format instead of clang-format, so that only new changes are re-formatted.